lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -1428,26 +1453,42 @@ fromMaryTx tx =
             Nothing
         }
     , map fromShelleyCert (toList certs)
-    , assetsToMint
-    , assetsToBurn
+    , TokenMapWithScripts assetsToMint mintScriptMap
+    , TokenMapWithScripts assetsToBurn burnScriptMap
     )
   where
-    SL.Tx bod _wits mad = tx
+    SL.Tx bod wits mad = tx
     MA.TxBody ins outs certs wdrls fee _valid _upd _adh mint = bod
     (assetsToMint, assetsToBurn) = fromLedgerMintValue mint
+    scriptMap = fromMaryScriptMap $ Shelley.scriptWits wits
+    (mintScriptMap, burnScriptMap)
+        | (assetsToMint == TokenMap.empty && assetsToBurn /= TokenMap.empty)
+        = (Map.empty, scriptMap)
+        | (assetsToMint /= TokenMap.empty && assetsToBurn == TokenMap.empty)
+        = (scriptMap, Map.empty)
+        | otherwise = (Map.empty, Map.empty)
```

Comment:
@paweljakubas What if the `mint` field contains **_both_** assets to mint **_and_** assets to burn?

It seems that in this case, we'd fall through to the `otherwise` case.

Is that expected?

According to my understanding, it's possible to **_both_** mint **_and_** burn assets in a single transaction:
- Assets to **_mint_** are represented by **_positive_** values in the `mint` map.
- Assets to **_burn_** are represented by **_negative_** values in the `mint` map.

The `fromLedgerMintValue` function separates these out into two `TokenMap` values, because `TokenMap` values can only contain non-negative quantities.

Reply:
added to investigate that in next PR:
see point 8 in description of this PR : `Look if simultaneous minting and burning works ok in Compatibility plus add integration test`